sentry-android-core/src/main/java/io/sentry/android/core/internal/measurement/cpu/CpuBackgroundMeasurementCollector.java

```diff
@@ -0,0 +1,110 @@
+package io.sentry.android.core.internal.measurement.cpu;
+
+import android.annotation.SuppressLint;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.android.core.internal.util.CpuInfoUtils;
+import io.sentry.measurement.MeasurementBackgroundCollector;
+import io.sentry.measurement.MeasurementBackgroundServiceType;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class CpuBackgroundMeasurementCollector implements MeasurementBackgroundCollector {
+
+  private final SentryOptions options;
+
+  public CpuBackgroundMeasurementCollector(@NotNull SentryOptions options) {
+    this.options = options;
+  }
+
+  @Override
+  public @NotNull MeasurementBackgroundServiceType getMeasurementType() {
+    return MeasurementBackgroundServiceType.CPU;
+  }
+
+  @Override
+  @SuppressLint("NewApi")
+  public @Nullable Object collect() {
+    try {
+      // TODO 1-5 ms
+      String stat = readProcSelfStat();
+      Long clockTicks = parseClockTicks(stat);
+
+      // TODO 2-4 ms on first read, then cached 4 - 20 μs
+      List<Integer> maxFrequencies = CpuInfoUtils.getInstance().readMaxFrequencies();
```

Comment:
Doesn't seem to return useful data on my emulator, need to wait for new device to arrive to retest.